### ISSUE TITLE
crypto(gov): submitting delegator vote proofs fail to verify

### ISSUE THREAD

**Issue by erwanor** [penumbra-zone/penumbra#2967]
**Describe the bug**
I am not able to submit a vote on any governance proposal on my local devnet because the delegator vote proof fails to verify.

```
$: cargo run --release --bin pcli -- -n http://127.0.0.1:8080 tx  vote yes --on 1
...
Error: status: Internal, message: "could not broadcast transaction: transaction pre-submission checks failed: a delegator vote proof did not verify: delegator vote proof did not verify", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc"} }
```

**To Reproduce**
1. Query the validator list
2. Delegate some penumbra to a validator
3. Create a signaling proposal: `cargo run --release --bin pcli -- tx proposal template signaling` and save it to `signal.toml`
4. Submit the proposal: `cargo run --release --bin pcli -- tx proposal submit --file signal.toml`
5. Check that the proposal is present: `cargo run --release --bin pcli -- query governance list-proposals`
6. Vote on the proposal: `cargo run --release --bin pcli -- tx vote yes --on 0`
7. See error

**Expected behavior**
The proof should not fail to verify during the pre-submission checks.

**Additional context**
It's possible that the proof uses the wrong balance commitment (unbonded amount or staked note) as input.

**Comment by erwanor** [comment]
Reproducing on `v0.59.0` leads to the following error:

```
Error: status: Unavailable, message: "error fetching notes: Invalid column name: asset_id", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc"} }
```

This was fixed by [907d6f854ddc98c829a0ea97ece4386f9e3b546b](https://github.com/penumbra-zone/penumbra/commit/907d6f854ddc98c829a0ea97ece4386f9e3b546b). You can do `git checkout 907d6f854ddc98c829a0ea97ece4386f9e3b546b` and run the commands with the fix.

**Comment by erwanor** [comment]
I'm checking that the public inputs to the proof match those that we feed to the verifier, and I noticed that on one hand (proving) we use the balance commitment to the unbonded amount and in the other we use the balance commitment to the staked note. I think that what we want to do is have a commitment to the voting power at the time of delegation (i.e. unbonded amount).

**Comment by erwanor** [comment]
Fixed in #2969
